File: src/main/java/tr1nks/service/logic/ParseService.java
package tr1nks.service.logic;

import org.springframework.web.multipart.MultipartFile;
import tr1nks.domain.dto.StudentDTO;

import javax.servlet.http.HttpSession;
import java.nio.charset.Charset;
import java.util.List;

public interface ParseService {

    boolean parseStudents(char delimiter, Charset charset, MultipartFile file, HttpSession session);

    boolean checkStudents(List<StudentDTO> studentDTOS, HttpSession httpSession);
}

File: src/main/java/tr1nks/service/logic/ParseService.kt
package tr1nks.service.logic

import org.springframework.web.multipart.MultipartFile
import tr1nks.domain.dto.GroupDTO
import tr1nks.domain.dto.StudentDTO
import java.nio.charset.Charset
import javax.servlet.http.HttpSession

interface ParseService {

    fun parseStudents(delimiter: Char, charset: Charset, file: MultipartFile, session: HttpSession): Boolean

    fun checkStudents(studentDTOS: List<StudentDTO>, httpSession: HttpSession): Boolean
}
